RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT.cc

```diff
@@ -0,0 +1,778 @@
+  //GlobalError dummyError; 
+
+  // ensure correct normalization of probabilities, should make double assginment reasonably impossible
+  for(unsigned int i=0; i<nt; i++){  
+    tks[i].Z=rho0*exp(-beta*( dzCutOff_*dzCutOff_));
+    for(vector<vertex_t>::iterator k=y.begin(); k!=y.end(); k++){ 
+      tks[i].Z += k->pk * exp(-beta*Eik(tks[i],*k));
+    }
+  }
+
+
+  for(vector<vertex_t>::iterator k=y.begin(); k!=y.end(); k++){ 
+    GlobalPoint pos(0, 0, k->z);
+    double time = k->t;
+    vector< reco::TransientTrack > vertexTracks;
+    //double max_track_time_err2 = 0;
+    double mean = 0.;
+    double expv_x2 = 0.;
+    double normw = 0.;
+    for(unsigned int i=0; i<nt; i++){
+      const double invdt = 1.0/std::sqrt(tks[i].dt2);
+      if(tks[i].Z>0){
+	double p = k->pk * exp(-beta*Eik(tks[i],*k)) / tks[i].Z;
+	if( (tks[i].pi>0) && ( p > 0.5 ) ){ 
+          //std::cout << "pushing back " << i << ' ' << tks[i].tt << std::endl;
+          vertexTracks.push_back(*(tks[i].tt)); tks[i].Z=0; 
+          mean     += tks[i].t*invdt*p;
+          expv_x2  += tks[i].t*tks[i].t*invdt*p;
+          normw    += invdt*p;
+        } // setting Z=0 excludes double assignment        
+      }
+    }
+    mean = mean/normw;
+    expv_x2 = expv_x2/normw;
+    const double time_var = expv_x2 - mean*mean;    
+    const double crappy_error_guess = std::sqrt(time_var);
+    GlobalError dummyErrorWithTime(0,
+                                   0,0,
+                                   0,0,0,
+                                   0,0,0,crappy_error_guess);
+    TransientVertex v(pos, time, dummyErrorWithTime, vertexTracks, 5);
+    clusters.push_back(v);
+  }
+
+
+  return clusters;
+
+}
+
+
+
+
+
+vector< vector<reco::TransientTrack> >
+DAClusterizerInZT::clusterize(const vector<reco::TransientTrack> & tracks)
+  const
+{
+  if(verbose_) {
+    cout << "###################################################" << endl;
+    cout << "# DAClusterizerInZT::clusterize   nt="<<tracks.size() << endl;
+    cout << "###################################################" << endl;
+  }
+
+  vector< vector<reco::TransientTrack> > clusters;
+  vector< TransientVertex > pv=vertices(tracks);
+
+  if(verbose_){ cout << "# DAClusterizerInZT::clusterize   pv.size="<<pv.size() << endl;  }
+  if (pv.size()==0){ return clusters;}
+
+
+  // fill into clusters and merge
+  vector< reco::TransientTrack>  aCluster=pv.begin()->originalTracks();
+  
+  if( verbose_ ) { 
+      std::cout << '\t' << 0;
+      std::cout << ' ' << pv.begin()->position() << ' ' << pv.begin()->time() << std::endl; 
+    }
+
+  for(vector<TransientVertex>::iterator k=pv.begin()+1; k!=pv.end(); k++){
+    if( verbose_ ) { 
+      std::cout << '\t' << std::distance(pv.begin(),k);
+      std::cout << ' ' << k->position() << ' ' << k->time() << std::endl; 
+    }
+    if ( std::abs(k->position().z() - (k-1)->position().z()) > (2*vertexSize_) ||
+         std::abs(k->time() - (k-1)->time()) > 2*0.010 ) {
```

Comment:
is 0.01 fundamental and unique here? 
Switch to a named constexpr or something derived from the config parameters